pallets/runtime/tests/src/confidential_asset_test.rs

```diff
@@ -15,31 +14,206 @@ use cryptography::{
     AssetId,
 };
 use curve25519_dalek::scalar::Scalar;
-use frame_support::assert_ok;
-use pallet_asset::{self as asset, AssetType, IdentifierType, SecurityToken};
+use frame_support::IterableStorageMap;
+use pallet_asset::{self as asset};
 use pallet_confidential_asset as confidential_asset;
-use polymesh_primitives::Ticker;
+use pallet_statistics as statistics;
 use rand::{rngs::StdRng, SeedableRng};
 use test_client::AccountKeyring;
 
-type ConfidentialAsset = confidential_asset::Module<TestStorage>;
+use frame_support::{assert_err, assert_ok, StorageDoubleMap, StorageMap};
+use pallet_identity as identity;
+use polymesh_primitives::{
+    AssetOwnershipRelation, AssetType, FundingRoundName, IdentifierType, SecurityToken, Ticker,
+};
+use sp_core::sr25519::Public;
+
+type Identity = identity::Module<TestStorage>;
+type Asset = asset::Module<TestStorage>;
+type AssetError = asset::Error<TestStorage>;
 type Origin = <TestStorage as frame_system::Trait>::Origin;
+type DidRecords = identity::DidRecords<TestStorage>;
+type Statistics = statistics::Module<TestStorage>;
+type System = frame_system::Module<TestStorage>;
+type ConfidentialAsset = confidential_asset::Module<TestStorage>;
+
+fn create_confidential_token(token_name: &[u8], ticker: Ticker, keyring: Public) {
+    assert_ok!(ConfidentialAsset::create_confidential_asset(
+        Origin::signed(keyring),
+        token_name.into(),
+        ticker,
+        100_000,
+        true,
+        AssetType::default(),
+        vec![],
+        None,
+    ));
+}
+
+#[test]
+fn issuers_can_create_and_rename_confidential_tokens() {
+    ExtBuilder::default().build().execute_with(|| {
+        let owner_signed = Origin::signed(AccountKeyring::Dave.public());
+        let owner_did = register_keyring_account(AccountKeyring::Dave).unwrap();
+        let funding_round_name: FundingRoundName = b"round1".into();
```

Comment:
```suggestion
        let funding_round_name = b"round1".into();
```
Shouldn't be necessary, as `create_confidential_asset` should constrain the type.

Reply:
apparently `into()` can't infer the type, unless I make the it explicit.

Reply:
Ah, that's probably because the method dispatch is forcing "we must know the type at this point" since rustc is using bidirectional type-checking instead of more constraint-solving-based unification.